### DISCUSSION TITLE
八股文如何背诵？

### DISCUSSION THREAD

**Discussion by hissenss** [freestylefly/CodeCanvas#25] [Q&A] [answered]
关于八股文背诵这方面请给我一些建议，我应该怎么做呢（真的好难受），可以和我说下背八股文的一些习惯啥的嘛？目前背的（集合，异常，redis，mysql，spring，加上接下来就是应该就是背星球整理的 项目相关的面经），还有什么要背的吗，

**Comment by freestylefly** [comment, marked as answer]
你好，第一次回答你的提问有点紧张，主要是我接下来的回答你可能觉得有些幼稚。

哈哈，抱歉以这种方式开头，只是想告诉你，向我提问没必要紧张，我通常都是尽自己努力来解答问题。

实话实说，你的这种背的方式很有可能在实际面试中记不住，而且会发现一个趋势，稍微有点经验的面试官都不会单纯的问你纯粹的八股，现阶段，对于初级工程师，企业最低的要求是一上来就能上手做项目，最好不用时间成本去培养就好了。特别是对于中小公司越发明显。那么面试问的多的可能会结合你的项目来问八股。

那该如何应对呢？

我建议以项目为出发点，去针对性的背八股，比如你做的某个项目有涉及到redis，可能只是用了redis的部分功能，但需要考虑八股中关于redis的知识，比如缓存穿透，缓存雪崩结合你的项目来思考。

比如你使用了redis怎么考虑缓存穿透问题的？你如果直接背你就会很懵逼根本记不住，那结合你自己的项目来理解呢？

你可以思考，我的项目中redis一旦挂了，大量数据直接落到数据库该怎么办？有什么技术可以解决，redis是缓存服务挂了缓存的数据还能在吗？要想持久化怎么搞。缓存和nacos配置如何同步如何配置持久化，这些问题你结合你的项目去思考去背会好很多。

但我希望是对每个技术的八股基础的还是绝对要啃一遍的，啃的过程是痛苦的，第一遍可以硬啃，第二遍结合项目来思考，会有很大帮助，我在最开始面试的时候也是这样啃的八股。

分享给你我的经验，他不一定适合所有人，但你可以尝试一下！

最后，祝好！
